src/analytics.js

```diff
@@ -0,0 +1,103 @@
+// @flow
+
+import fs from 'fs-extra'
+import { default as OS } from 'os'
+// flow$ignore
+import type Config from 'cli-engine-config'
+import HTTP from 'cli-engine-command/lib/http'
+import Netrc from 'netrc-parser'
+import Output from 'cli-engine-command/lib/output'
+import URI from 'url'
+
+export default class AnalyticsCommand {
+  command: string
+  plugin: string
+  pluginVersion: string
+  version: string
+  platform: string
+  arch: string
+  start: any
+  config: Config
+
+  constructor (command: string, plugin: string, pluginVersion: string, config: Config) {
+    this.config = config
+    this.command = command
+    this.plugin = plugin
+    this.pluginVersion = pluginVersion
+    this.platform = config.platform
+    this.arch = OS.arch()
+    this.version = this.config.version
+  }
+
+  async record () {
+    if (AnalyticsCommand.skipAnalytics() || process.argv.length < 2) {
+      return
+    }
+    let analyticsJSON
+    if (fs.existsSync(AnalyticsCommand.analyticsPath(this.config))) {
+      let analyticsData = fs.readFileSync(AnalyticsCommand.analyticsPath(this.config), 'utf8')
+      analyticsJSON = JSON.parse(analyticsData)
+    } else {
+      analyticsJSON = {
+        'schema': 1,
+        'commands': [],
+        'user': 'not specified'
+      }
+      const netrcUser = await AnalyticsCommand.netrcLogin()
+      if (netrcUser) analyticsJSON['user'] = netrcUser
+    }
+    analyticsJSON['commands'].push({
+      command: this.command,
+      version: this.config.version,
+      platform: this.platform
+    })
+    fs.writeFileSync(AnalyticsCommand.analyticsPath(this.config), JSON.stringify(analyticsJSON), 'utf8')
+  }
+
+  static async submitAnalytics (config: Config) {
```

Comment:
why is this static but the record function is an instance method? seems like it would be better for them to both be instance methods